### ISSUE TITLE
Remove Android Pay from the Stock package

### ISSUE THREAD

**Issue by DJAlik** [opengapps/opengapps#116]
With Android Pay not working on rooted devices, and I am assuming everyone using opengapps is rooted, we should probably remove it from the package. Some discussion on this topic here: http://www.androidpolice.com/2015/09/25/google-security-engineer-visits-xda-forums-to-explain-why-android-pay-doesnt-support-rooted-devices/

**Comment by mfonville** [comment]
I think it is best to keep it in for now, because it is useful for those who are flashing 'For Stock' style.
I still have to read the article you did link, but I personally do hope that at some point a solution will be found and custom and rooted ROMs will be allowed too.

**Comment by DJAlik** [comment]
Based on the XDA thread linked in the article the Google engineer is not
leaving much hope for that.

On Fri, Sep 25, 2015 at 4:47 PM, mfonville <notifications@github.com> wrote:

> I think it is best to keep it in for now, because it is useful for those
> who are flashing 'For Stock' style.
> I still have to read the article you did link, but I personally do hope
> that at some point a solution will be found and custom and rooted ROMs will
> be allowed too.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opengapps/opengapps/issues/116#issuecomment-143361588>
> .
>



-- 
<http://www.facebook.com/djalik>  <http://twitter.com/#!/DJAlik>
<http://www.linkedin.com/in/alexrusinov>
<http://plus.google.com/114580313768137070684>
<http://www.mixcloud.com/djalik/>
<http://psnprofiles.com/DJAlik>

**Comment by mfonville** [comment]
I am wondering btw if flashing 'for stock' gapps style does break the Safetynet API.
@bgiesing I believe you have have a device with "stock" ROM and living in the USA. Could you maybe test this for us?

**Comment by rapperskull** [comment]
Wait a minute. I found this: https://www.reddit.com/r/Android/comments/3kly2z/checking_device_compatibility_with_safetynet/ on Reddit, and there are some useful informations:
1. One user said:
*Been messing with stuff all day, some things I've discovered:
-Unlocked bootloader doesn't matter.
-Can't have root installed initially.
-Has to be a stock / signed ROM.
I attempted to see what would happen if I setup Android Pay on the stock version of CM12.1 for OPO. I was able to set it up successfully. I then backed up GPS and Android Pay with Titanium Backup and restored my Unofficial CM12.1 Nightlies ROM. I restored the data and restarted. Upon launching Android Pay, I saw my fully setup account with my 3 credit cards for a split second before the 'Add a Card' window displayed.*
2. Another one:
*Okey, I disabled this check (better said, faked to true) via Xposed.
I ran SafetyNet example app from Store, got CTS profile match:false. After I applied my module, it is true and app is happy now.
I want to pack it into my XInstaller module (it has really advanced options). Maybe later I will do standalone module (want it?)
p.s: it was so simple. The simpliest solution is the best one!*

Source code of the module: https://github.com/pylerSM/NoDeviceCheck/blob/master/src/com/pyler/nodevicecheck/NoDeviceCheck.java
Available in the Xposed Repository too.


**tl;dr:
Maybe there's hope.**

**Comment by bgiesing** [comment]
@mfonville I had to switch devices and the new one I have can't be unlocked or rooted meaning no custom recovery to even flash.

**Comment by mfonville** [comment]
@bgiesing pity, then does somebody else @opengapps/general have a device in the USA that they could test with Android Pay with a Stock ROM + 'For Stock' updates flashed?

**Comment by mfonville** [comment]
Since I don't see any actitivity in that reddit thread anymore and the XDA thread is closed.
And I am really sure that flashing 'for nexus'-style on a stock ROM would also trigger a failure to pass the SafetyNet test (since system files will have changed) I think, that at this very moment, we should drop the AndroidPay package from the builds.
We can keep all references to the script in place, even keep it updated in the sources, but it just doesn't make sense to bundle it until a solution is found to run it on Custom ROMs.

**Comment by DJAlik** [comment]
should we replace it with the Google Wallet app?

On Sun, Oct 4, 2015 at 11:59 AM, mfonville <notifications@github.com> wrote:

> Since I don't see any actitivity in that reddit thread anymore and the XDA
> thread is closed.
> And I am really sure that flashing 'for nexus'-style on a stock ROM would
> also trigger a failure to pass the SafetyNet test (since system files will
> have changed) I think, that at this very moment, we should drop the
> AndroidPay package from the builds.
> We can keep all references to the script in place, even keep it updated in
> the sources, but it just doesn't make sense to bundle it until a solution
> is found to run it on Custom ROMs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opengapps/opengapps/issues/116#issuecomment-145366536>
> .
>



-- 
<http://www.facebook.com/djalik>  <http://twitter.com/#!/DJAlik>
<http://www.linkedin.com/in/alexrusinov>
<http://plus.google.com/114580313768137070684>
<http://www.mixcloud.com/djalik/>
<http://psnprofiles.com/DJAlik>

**Comment by mfonville** [comment]
No, that would break things down.

**Comment by mfonville** [comment]
Done, by creation of the super package.
